tests/datasets/test_transfer.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2018-2019 Faculty Science Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import random
+import string
+from uuid import uuid4
+
+import pytest
+
+from faculty.clients.object import CloudStorageProvider, CompletedUploadPart
+from faculty.datasets import transfer
+
+
+PROJECT_ID = uuid4()
+TEST_PATH = "/path/to/file"
+TEST_URL = "https://example.com/presigned/url"
+OTHER_URL = "https://example.com/other-presigned/url"
+TEST_AWS_UPLOAD_ID = 123
+
+TEST_ETAG = "5d24e152bcdfa5a0357f46471be3be6c"
+TEST_COMPLETED_PART = CompletedUploadPart(1, TEST_ETAG)
+
+OTHER_ETAG = "d084dd881a190aa5ffdf0ce21cff9509"
+OTHER_COMPLETED_PART = CompletedUploadPart(2, OTHER_ETAG)
+
+TEST_CONTENT = "".join(
+    random.choice(string.printable) for _ in range(2000)
+).encode("utf8")
+
+
+@pytest.fixture
+def mock_client_download(mocker, requests_mock):
+    object_client = mocker.Mock()
+    object_client.presign_download.return_value = TEST_URL
+
+    requests_mock.get(TEST_URL, content=TEST_CONTENT)
+
+    yield object_client
+
+    object_client.presign_download.assert_called_once_with(
+        PROJECT_ID, TEST_PATH
+    )
+
+
+@pytest.fixture
+def mock_client_upload_s3(mocker, requests_mock):
+    presigned_response_mock = mocker.Mock()
+    presigned_response_mock.provider = CloudStorageProvider.S3
+    presigned_response_mock.upload_id = TEST_AWS_UPLOAD_ID
+
+    object_client = mocker.Mock()
+    object_client.presign_upload.return_value = presigned_response_mock
+
+    yield object_client
```

Comment:
If we don't want this fixture to do anything after the test finishes, it does not need to be a generator and we can just:
```suggestion
    return object_client
```
However, we may want to keep it as a generator and add the following:
```python
object_client.presign_upload.assert_called_once_with(PROJECT_ID, TEST_PATH)
```

Reply:
I think I just forgot the assertion. I'll add that now